src/store/data-store-level.ts

```diff
@@ -85,22 +133,32 @@ export class DataStoreLevel implements DataStore {
     return readableStream;
   }
 
-  public async has(tenant: string, recordId: string, dataCid: string): Promise<boolean> {
-    const partition = await this.blockstore.partition(tenant);
+  public async delete(tenant: string, recordId: string, dataCid: string): Promise<void> {
+    const recordForCid = await this.blockstore.partition(HOST_PARTITION);
+    const tenantForRecord = await recordForCid.partition(dataCid);
+    const dataForTenant = await tenantForRecord.partition(recordId);
+    const tenants = await dataForTenant.partition(tenant);
```

Comment:
sanity: the plural `tenants` currently throws me off, as `dataForTenant.partition(tenant)` already selects a partition base on a tenant, why would this be plural, same question mark for occurrences in above methods. Thus why would `tenants.isEmpty()` ever be `false` on line 144 after `tenants.delete(tenant)? Code coverage result also hints at the same thing as the `if (!wasLastTenant)` condition being `true` is never covered, or maybe that's coincidental and that we are just missing (thus ought to add) a test to cover the case of deleting data that is referenced in multiple tenants, if the test doesn't exist already. Will keep reading.

Reply:
oh lol yup im pretty sure we can just remove that

i think it was a leftover from an earlier design 😅